pallets/pooled-staking/src/pools.rs

```diff
@@ -0,0 +1,396 @@
+        .checked_add(&leaving)
+        .ok_or(Error::InconsistentState)?;
+
+    ensure!(total0 == total1, Error::InconsistentState);
+
+    Ok(())
+}
+
+macro_rules! impl_pool {
+    ($name:ident, $shares:ident, $supply:ident, $total:ident, $hold: ident, $init:expr $(,)?) => {
+        pub struct $name<T>(PhantomData<T>);
+        impl<T: Config> Pool<T> for $name<T> {
+            fn shares(candidate: &Candidate<T>, delegator: &Delegator<T>) -> Shares<T> {
+                Shares(Pools::<T>::get(
+                    candidate,
+                    &PoolsKey::$shares {
+                        delegator: delegator.clone(),
+                    },
+                ))
+            }
+
+            fn shares_supply(candidate: &Candidate<T>) -> Shares<T> {
+                Shares(Pools::<T>::get(candidate, &PoolsKey::$supply))
+            }
+
+            fn total_staked(candidate: &Candidate<T>) -> Stake<T> {
+                Stake(Pools::<T>::get(candidate, &PoolsKey::$total))
+            }
+
+            fn hold(candidate: &Candidate<T>, delegator: &Delegator<T>) -> Stake<T> {
+                Stake(Pools::<T>::get(
+                    candidate,
+                    &PoolsKey::$hold {
+                        delegator: delegator.clone(),
+                    },
+                ))
+            }
+
+            fn set_shares(candidate: &Candidate<T>, delegator: &Delegator<T>, value: Shares<T>) {
+                Pools::<T>::set(
+                    candidate,
+                    &PoolsKey::$shares {
+                        delegator: delegator.clone(),
+                    },
+                    value.0,
+                )
+            }
+
+            fn set_shares_supply(candidate: &Candidate<T>, value: Shares<T>) {
+                Pools::<T>::set(candidate, &PoolsKey::$supply, value.0)
+            }
+
+            fn set_total_staked(candidate: &Candidate<T>, value: Stake<T>) {
+                Pools::<T>::set(candidate, &PoolsKey::$total, value.0)
+            }
+
+            fn set_hold(candidate: &Candidate<T>, delegator: &Delegator<T>, value: Stake<T>) {
+                Pools::<T>::set(
+                    candidate,
+                    &PoolsKey::$hold {
+                        delegator: delegator.clone(),
+                    },
+                    value.0,
+                )
+            }
+
+            fn initial_share_value() -> Stake<T> {
+                Stake($init)
+            }
+        }
+    };
+}
+
+impl_pool!(
+    Joining,
+    JoiningShares,
+    JoiningSharesSupply,
+    JoiningSharesTotalStaked,
+    JoiningSharesHeldStake,
+    T::Balance::one(),
+);
+
+impl_pool!(
+    AutoCompounding,
+    AutoCompoundingShares,
+    AutoCompoundingSharesSupply,
+    AutoCompoundingSharesTotalStaked,
+    AutoCompoundingSharesHeldStake,
+    T::InitialAutoCompoundingShareValue::get(),
+);
+
+impl_pool!(
+    ManualRewards,
+    ManualRewardsShares,
+    ManualRewardsSharesSupply,
+    ManualRewardsTotalStaked,
```

Comment:
This name is inconsistent with `AutoCompoundingSharesTotalStaked`

```suggestion
    ManualRewardsSharesTotalStaked,
```

Reply:
Fixed